packages/cozy-harvest-lib/src/components/OAuthForm.jsx

```diff
@@ -1,19 +1,81 @@
 import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
 
 import Button from 'cozy-ui/react/Button'
 import { translate } from 'cozy-ui/react/I18n'
 
+import OAuthWindow from './OAuthWindow'
+
+const IDLE = 'idle'
+const WAITING = 'waiting'
+
+/**
+ * The OAuth Form is responsible for displaying a form for OAuth konnectors. It
+ * starts the OAuth process
+ */
 export class OAuthForm extends PureComponent {
+  state = {
+    status: IDLE
+  }
+
+  constructor(props, context) {
+    super(props, context)
+    this.endOAuth = this.endOAuth.bind(this)
+    this.handleAccountId = this.handleAccountId.bind(this)
+    this.startOAuth = this.startOAuth.bind(this)
+  }
+
+  endOAuth() {
+    this.setState({ status: IDLE })
+  }
+
+  handleAccountId(accountId) {
+    const { onSuccess } = this.props
+    if (typeof onSuccess === 'function') onSuccess(accountId)
+  }
+
+  startOAuth() {
+    this.setState({ status: WAITING })
+  }
+
   render() {
-    const { initialValues, t } = this.props
+    const { konnector, initialValues, submitting, t } = this.props
+    const { status } = this.state
+    const isWaiting = status === WAITING
+    const isBusy = isWaiting || submitting
     return initialValues ? null : (
-      <Button
-        className="u-mt-1"
-        extension="full"
-        label={t('oauth.connect.label')}
-      />
+      <>
+        <Button
+          className="u-mt-1"
+          busy={isBusy}
+          disabled={isBusy}
+          extension="full"
+          label={t('oauth.connect.label')}
+          onClick={this.startOAuth}
+        />
+        {isWaiting && (
+          <OAuthWindow
+            konnector={konnector}
+            onAccountId={this.handleAccountId}
+            onCancel={this.endOAuth}
+          />
+        )}
+      </>
     )
   }
 }
 
+OAuthForm.propTypes = {
+  /** Related Konnector */
+  konnector: PropTypes.object.isRequired,
+  /** Initial values, tipically when an account already exists */
+  initialValues: PropTypes.object,
+  /** Success Callback, take account as parameter */
+  onSuccess: PropTypes.func,
+  /** Indicates if the form should be rendered as submitting */
+  submitting: PropTypes.bool,
```

Comment:
I am intrigued, I would have thought the form would manage its state concerning submitting.

Reply:
In Cozy-Home, we may render the form with a job already running. Maybe `busy` should be a better prop name.

Reply:
OK I do not have an immediate alternative ;)